src/pip/_internal/commands/show.py

```diff
@@ -155,6 +155,11 @@ def _files_from_legacy(dist: BaseDistribution) -> Optional[Iterator[str]]:
         except KeyError:
             continue
 
+        requires = sorted(
+            (req.name for req in dist.iter_dependencies()), key=lambda pkg: pkg.lower()
+        )
+        required_by = sorted(_get_requiring_packages(dist), key=lambda pkg: pkg.lower())
```

Comment:
Use `operator.methodcaller("lower") instead.

Also, since `_get_requiring_packages` not longer needs return a list, please change it to return an iterator instead.

Reply:
We could even use `key=str.lower` here.

Reply:
@uranusjr, good point. I have converted it to an iterator.

@pradyunsg, I didn't think of that. It's definitely nicer than a lambda. I have changed it.

On a side note, I don't really know what the development process is here. Should I squash and rebase once this is approved and before being merged, or is that not something I need to worry about?